src/Streaming/Cassava.hs

```diff
@@ -126,9 +126,12 @@ runParser :: (Monad m) => Parser a -> ByteString m r
              -> Stream (Of (Either CsvParseException a)) m (Either (CsvParseException, ByteString m r) r)
 runParser = loop
   where
-    feed f str = (uncurry (loop . f) . fromMaybe (mempty, str))
-                          -- nxt == Nothing, str is just Return
-                 =<< lift (B.unconsChunk str)
+    feed f str = do
+      nxt <- lift (B.nextChunk str)
+      let g = loop . f
```

Comment:
I don't think this is really needed as it's only used one.

Reply:
I agree here. Perhaps I can create a few tests to make sure it really isn't needed.

Reply:
Thanks for the bug fix, there is and was no problem with `readFile`, rather the issue is that streams are in essence linear types, and once you've queried the stream, you MUST NOT use the original stream object again, rather the stream possibly returns some data and the "rest" of the stream, and one may at that point only use the "rest" of the stream, the original stream handle would attempt to perform the same I/O action twice, which is a no-no.

The changes in this PR avoid re-using the original `str` one more time with the `fromMaybe (mempty, str)` idioms, which are incorrect.  This PR should be released.

Also, separately, some CPP guards should be added and a bit of new code so that this code will build against `streaming-bytestring` 2.0 (which has some breaking changes).  Let me know if you need help with that.

Finally, is this package still maintained?  If not perhaps some new maintainers or co-maintainers would be welcome?

Reply:
Hi @vdukhovni,

I'm not sure what you're saying here; should this be merged in or shouldn't it?

Reply:
Nevermind, I see what you mean now.

Reply:
Thanks for merging, much appreciated. This PR is sufficient cause for a release.
But if want to reduce release churn, there could be a new PR to address bitrot (in particular compatibility with streaming-bytestring 2.0).  How would you like to proceed?

Also, would you prefer to tackle the update yourself, or to have it contributed for your review?

Reply:
One more thing, the disclaimer about `readFile` in the module documentation should now be removed...

Reply:
Done; version 0.1.0.2 has been released and as soon as CI passes I'll release 0.2.0.0 that supports streaming-bytestring-0.2.* as well.

Reply:
Note, there are some breaking changes in 2.0, so some code changes will be required, mostly around nextChunk/unconsChunk, but it is best to migrate to the new type names `ByteString -> ByteStream` with conditional type aliases for legacy releases. And the preferred module import names have changed...  It is a non-trivial port.

See https://github.com/archaephyrryx/jparse/tree/master/src/Data/ByteString/Streaming for the sort of changes required.